src/security/cert-storage.ts

```diff
@@ -135,17 +136,92 @@ export class CertStorage implements SecurityAgent {
     storage: Storage,
     fw: Forwarder,
     prvKeyBits: Uint8Array,
-    interestLifetime = 5000,
   ) {
     const result = new CertStorage(
       trustAnchor,
       ownCertificate,
       storage,
       fw,
       prvKeyBits,
-      interestLifetime,
     );
     await result.readyEvent;
     return result;
   }
+
+  /**
+   * Modified version of getCertificate function, designed for trying to
+   * fetch PoR from all other already trusted parties
+   * @param strangerKeyname First part of PoR name, i.e. "/<𝑍𝑏 name>/KEY/<𝑇𝑏 keyid>"
+   */
+  async fetchPoR(strangerKeyname: string) {
+    // loop through trusted certificates
+    for (const item of this.trustedNames) {
+      // Construct <Z_a name encoded>
+      const parts: string[] = item.split('/');
+      let partsBeforeKey: string = '';
+      for (let i = 0; i < parts.length; i++) { //put <Z_a name encoded> together with a loop
+        if (parts[i].includes('KEY')) {
+          break;
+        }
+        partsBeforeKey += parts[i] + '/';
+      }
```

Comment:
First, I don't think this is the correct behavior. I think `<Tb keyid>` is a **single** component containing the encoded name. Please confirm with @tianyuan129 .
If this behavior is correct, you can still get rid of the loop. See [Name.append](https://ndnts-docs.ndn.today/typedoc/classes/_ndn_packet.name-1#append)